bot/cogs/moderation.py

```diff
@@ -875,121 +884,68 @@ def mod_log(self) -> ModLog:
         await ctx.invoke(self.bot.get_command("help"), "infraction")
 
     @with_role(*MODERATION_ROLES)
-    @infraction_group.group(name='edit', invoke_without_command=True)
-    async def infraction_edit_group(self, ctx: Context) -> None:
-        """Infraction editing commands."""
-        await ctx.invoke(self.bot.get_command("help"), "infraction", "edit")
-
-    @with_role(*MODERATION_ROLES)
-    @infraction_edit_group.command(name="duration")
-    async def edit_duration(
-            self, ctx: Context,
-            infraction_id: int, expires_at: Union[Duration, str]
+    @infraction_group.command(name='edit')
+    async def infraction_edit(
+        self,
+        ctx: Context,
+        infraction_id: int,
+        expires_at: Union[Duration, permanent_duration, None],
```

Comment:
Too bad we can't really mix "just" comments and requested changes, so I've added this as a separate comment.

Yesterday (for me) we briefly talked about being able to set exact dates instead of a relative duration for the expiration field. I've checked and I've not seen it used, so I'm not married to it. If we do want it, I may be better to have a separate converter that converts datetime strings of a certain format to a datetime instead of creating a hybrid one that both parses relative durations and absolute datetime strings.

I currently see not no real use for it, though, although it could be useful if we ever want to do something with absolute dates (like adding events to the event calendar we've envisioned.)

Reply:
Exact dates could be handy for underage bans.

Reply:
> may be better to have a separate converter that converts datetime strings of a certain format to a datetime instead of creating a hybrid one that both parses relative durations and absolute datetime strings.

I agree. I came to that conclusion too as I started working on this. I'm not sure if this would be good to implement in this PR or in another.